utils/dockerutil.py

```diff
@@ -95,3 +102,39 @@ def find_cgroup_filename_pattern(mountpoints, container_id):
             return os.path.join('%(mountpoint)s/system/docker/%(id)s/%(file)s')
 
     raise MountException("Cannot find Docker cgroup directory. Be sure your system is supported.")
+
+
+def image_tag_extractor(co, key):
+    if "Image" in co:
+        split = co["Image"].split(":")
+        if len(split) <= key:
+            return None
+        elif len(split) > 2:
+            # if the repo is in the image name and has the form 'docker.clearbit:5000'
+            # the split will be like [repo_url, repo_port/image_name, image_tag]. Let's avoid that
+            split = [':'.join(split[:-1]), split[-1]]
+        return [split[key]]
+    if "RepoTags" in co:
+        splits = [el.split(":") for el in co["RepoTags"]]
+        tags = []
+        for split in splits:
+            if len(split) > 2:
+                split = [':'.join(split[:-1]), split[-1]]
+            if len(split) > key:
+                tags.append(split[key])
+        if len(tags) > 0:
+            return list(set(tags))
+    return None
+
+
+def container_name_extractor(co):
+    names = co.get('Names', [])
+    if names is not None:
+        # we sort the list to make sure that a docker API update introducing
+        # new names with a single "/" won't make us report dups.
+        names = sorted(names)
+        for name in names:
+            # the leading "/" is legit, if there's another one it means the name is actually an alias
+            if name.count('/') <= 1:
+                return [str(name).lstrip('/')]
+    return co.get('Id')[:11]
```

Comment:
Because the short ID version of IDs consists in the first 12 digits of the ID. You can see it from most the Docker CLI commands, but it is also supported as ID from all the CLI commands.

So basically, that is the real identifier people see.  To see the full ID, you have to inspect or use the API.